variantlib/models/variant.py

```diff
@@ -127,14 +129,16 @@ def to_str(self) -> str:
     @classmethod
     def from_str(cls, input_str: str) -> Self:
         # removing starting `^` and trailing `$`
-        pttn_nmspc = VALIDATION_NAMESPACE_REGEX[1:-1]
-        pttn_feature = VALIDATION_FEATURE_REGEX[1:-1]
-        pttn_value = VALIDATION_VALUE_REGEX[1:-1]
+        pttn_nmspc = VALIDATION_NAMESPACE_REGEX.pattern[1:-1]
+        pttn_feature = VALIDATION_FEATURE_REGEX.pattern[1:-1]
+        pttn_value = VALIDATION_VALUE_REGEX.pattern[1:-1]
```

Comment:
Hmm… I think we can drop the `^` and `$` bits out of regex and use `.fullmatch()` where necessary instead. But let's look into this separately, I'll add it to #23.